include/Scope.h

```diff
@@ -1,36 +1,44 @@
 #pragma once
 
+#include <vector>
 #include <unordered_map>
 
 #include "StringRef.h"
 
 namespace slang {
 
 class Symbol;
+enum class SymbolKind;
 
 // Maintains context (like in a specific scope) and knows about parent
 // contexts so that symbol lookup can be performed.
 class Scope {
 public:
-    Scope() {};
+    Scope();
 
-    void add(const Symbol* symbol);
+    bool add(const Symbol* symbol);
 
     template<typename TContainer>
-    void addRange(const TContainer& container) {
+    bool addRange(const TContainer& container) {
```

Comment:
This is fine for now, but it's hard to know what to do if we get a false return from this function because we don't know which symbols were added and which weren't.